src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -1446,6 +1455,13 @@ _public_ sd_bus* sd_bus_flush_close_unref(sd_bus *bus) {
         if (!bus)
                 return NULL;
 
+        /* Have to do this before flush() to prevent hang */
+        if (bus->busexec_pid)
+                if (kill(bus->busexec_pid, SIGTERM) == bus->busexec_pid) {
+                        (void)waitpid(bus->busexec_pid, NULL, 0);
+                        bus->busexec_pid = 0;
+                }
+
```

Comment:
given that we need this at least twice we should really have a helper function for this: "bus_kill_exec()" or so...

Reply:
also, this new function needs to be called in bus_start_address() too, where we close the fds from the previous try... i.e. next to the invocations of bus_close_io_fds()